pkg/frontend/middleware/audit.go

```diff
@@ -0,0 +1,134 @@
+package middleware
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/log"
+	"github.com/Azure/ARO-RP/pkg/util/log/audit"
+)
+
+// Audit generates an audit log based on the request, caller, resource and
+// correlation data found in a HTTP request. It depends on the 'Log' middleware
+// to populate the request context with data that it needs.
+func Audit(env env.Core, entry *logrus.Entry) func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// skip requests to the 'operationsstatus' endpoints
+			if strings.Contains(r.URL.Path, "operationsstatus") {
+				h.ServeHTTP(w, r)
+				return
+			}
+
+			callerIdentity, callerType, correlationID, requestID := callerRequestData(r)
+			targetResourceName, targetResourceType := targetResourceData(entry, r)
+
+			auditEntry := audit.NewEntry(entry.Logger)
```

Comment:
Are you sure that entry.Logger is what's wanted?  This will give you a logger that was originally created by log.GetLogger() rather than log.GetBaseLogger() - with SetReportCaller, CallerPrettyfier and SetLevel set.  I am supposing that you would need to create the audit logger in cmd/ and pass it all the way through.